### DISCUSSION TITLE
Help Request: How to enable masquerade mode?

### DISCUSSION THREAD

**Discussion by undermyumbrella1** [TEAMMATES/teammates#12433] [Help requests] [answered]
### Environment

- master branch
- jdk 17
- Chrome Latest Version

### Intention

- Following the documentation
- logged in as user
- use masquerade mode by accessing `?user=tom`
- gets this error message
![image](https://github.com/TEAMMATES/teammates/assets/120079323/cd51584b-7837-488f-b3be-d49c649db222)

### Resolution Attempt

- upon tracing the code
- it seems masquerade user tom
- is not a Instructor
- the `isInstructor` flag is set in `UserProvision.java`
```
  /**
     * Gets the information of the current masqueraded user.
     */
    public UserInfo getMasqueradeUser(String googleId) {
        UserInfo userInfo = new UserInfo(googleId);
        userInfo.isAdmin = false;
        userInfo.isInstructor = instructorsLogic.isInstructorInAnyCourse(googleId);
        userInfo.isStudent = studentsLogic.isStudentInAnyCourse(googleId);
        userInfo.isMaintainer = Config.APP_MAINTAINERS.contains(googleId);
        return userInfo;
    }
```
- as the masquerade mode google id `tom` is not found in db, it will never be a instructor

- all create instructors have google id `app_admin@gmail.com`
- so masquerade mode with `?user=app_admin@gmail.com` is successful
![image](https://github.com/TEAMMATES/teammates/assets/120079323/8df2fbb0-1175-41b3-be20-92280c413d50)

How do i masquerade as another instructor i have created? (all created instructors have google id app_admin@gmail.com)

**Comment by domlimm** [comment]
You could try to get the sign up link via the terminal where you first ran `./gradlew serverRun`. Copy that link and place it into an incognito/another browser.

**Reply by undermyumbrella1** [reply, marked as answer]
it works, thank you!
Just to summarize, the alternative steps to masquerade:
1. Log in to /web/admin/home as an administrator
2. In admin pages, create new instructor with any Name, Email, Institution credentials
3. After adding instructor, do not access the join link in the same browser
4. Copy the join link into another browser/incognito browser, and register with the new instructor's email 
(instead of the default admin email)
6. Now there is an instructor with its own google id instead of the admin google id
7. This google id can be used to login as the new instructor

**Reply by domlimm** [reply]
Nice, exactly! Glad you got it sorted!

**Comment by domlimm** [comment]
Do create more discussion threads if you ever run into issues 👍🏻
